### PR TITLE
Fixes #20754: Ansible collection does not properly setup repository for (yet?) unsupported OS

### PR DESCRIPTION
**Pull request by julienbriault**
https://issues.rudder.io/issues/20754

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -182,15 +183,16 @@ and installs the packages.
...
 ```
 
 ### rudder_server
 
 #### Role variables
 
-- `server_version`: Rudder version(default: `6.2`)
+- `server_version`: Rudder version(default: `7.0`)
 - `rudder_repository`: Rudder repository domain (default: `repository.rudder.io`)
+- `rudder_repository_url`: Specific Rudder repository URL (default: `empty`), used only when is not empty.
```
Comment (Fdall) [inline README.md:195]:
```suggestion
- `rudder_repository_url`: Complete Rudder repository URL (default: `empty`), used only when not empty, replace the server_version and rudder_repository when used.
```

File: README.md [hunk 2]
```diff
@@ -241,8 +243,9 @@ the roles listed above.
 
 #### Role variables
 
-- `rudder_version`: Rudder version(default: `6.2`)
+- `rudder_version`: Rudder version(default: `7.0`)
 - `repository`: Rudder repository domain (default: `repository.rudder.io`)
+- `repository_url`: Specific Rudder repository URL (default: `empty`), used only when is not empty.
```
Comment (Fdall) [inline README.md:248]: Apply same change than above

File: README.md [hunk 3]
```diff
@@ -260,7 +263,7 @@ the roles listed above.
       roles:
         - role: rudder.rudder.rudder_repository
           vars:
-            rudder_version: 6.2.10
+            rudder_version: 7
```
Comment (Fdall) [inline README.md:266]:
```suggestion
            rudder_version: 7.0
```

**Comment by Fdall** [comment]
i am a bit surprise the rudder_repository_url is never passed to the sub role, how does it work?

**Comment by Fdall** [comment]
Can you add an error case for the repository setup if none of the supported OS are matched, just to signal to the user that it did not work as intended?

**Comment by julienbriault** [comment]
> i am a bit surprise the rudder_repository_url is never passed to the sub role, how does it work?

The placement of the rests must be done in advance. So you have to call "rudder_repository" once to set up the repos as desired and then launch the deployment.

We could make it call by the playbooks but the parameters would have to be modified directly in the collection files, something we don't want once it is installed.

**Comment by julienbriault** [comment]
> Can you add an error case for the repository setup if none of the supported OS are matched, just to signal to the user that it did not work as intended?

If the OS are not supported, Ansible automatically returns an error. I'm not sure if it's useful to specify it in hardcopy.

**Comment by Fdall** [comment]
> > Can you add an error case for the repository setup if none of the supported OS are matched, just to signal to the user that it did not work as intended?
> 
> If the OS are not supported, Ansible automatically returns an error. I'm not sure if it's useful to specify it in hardcopy.

No you don't, the roles are using a dependency system which make them able to call each other without any intervention of the user. See https://github.com/Normation/rudder-ansible/blob/9b1049e836d4da8a67ce9b400a73be9048627a49/roles/rudder_server/meta/main.yml#L34

**Comment by julienbriault** [comment]
> > > Can you add an error case for the repository setup if none of the supported OS are matched, just to signal to the user that it did not work as intended?
> > 
> > 
> > If the OS are not supported, Ansible automatically returns an error. I'm not sure if it's useful to specify it in hardcopy.
> 
> No you don't, the roles are using a dependency system which make them able to call each other without any intervention of the user. See
> 
> https://github.com/Normation/rudder-ansible/blob/9b1049e836d4da8a67ce9b400a73be9048627a49/roles/rudder_server/meta/main.yml#L34

File: roles/rudder_relay/meta/main.yml [hunk 4]
```diff
@@ -16,25 +16,43 @@ galaxy_info:
...
+         - 7
+         - 8
 
 dependencies:
   - role: rudder_repository
     vars:
       repository: "{{ rudder_repository }}"
+      rudder_version: "{{ relay_version }}"
+      repository_url: " {{ rudder_repository_url }}"
+      rudder_update_cache: " {{ update_cache }}"
```
Comment (Fdall) [inline roles/rudder_relay/meta/main.yml:54]: Why the space before?

File: roles/rudder_repository/tasks/debian.yml [hunk 5]
```diff
@@ -14,4 +14,14 @@
...
     update_cache: "{{ update_cache }}"
     validate_certs: yes
+  when: repository_url|length == 0
+  become: true
+
+- name: Adding Rudder Repo (from repository_url parameter)
+  apt_repository:
+    repo: "deb {{ repository_url }} main"
+    state: present
+    update_cache: "{{ update_cache }}"
```
Comment (Fdall) [inline roles/rudder_repository/tasks/debian.yml:24]: Why use "update_cache" here, if I read correctly it should be `rudder_update_cache` no ? it is the same everywhere

**Comment by julienbriault** [comment]
Everything has been corrected as recommended.

File: roles/rudder_server/meta/main.yml [hunk 6]
```diff
@@ -17,25 +17,43 @@ galaxy_info:
...
+         - 6
+         - 7
+         - 8
 
 dependencies:
   - role: rudder_repository
     vars:
       repository: "{{ rudder_repository }}"
+      rudder_version: "{{ server_version }}"
+      repository_url: " {{ rudder_repository_url }}"
```
Comment (Fdall) [inline roles/rudder_server/meta/main.yml:54]: there are still unwanted spaces here

**Pull request merged**